### PR TITLE
Finished ovl_Demo_Go

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_Go/z_demo_go.c [hunk 1]
```diff
@@ -200,21 +200,16 @@ UNK_TYPE DemoGo_FrameUpdateMatrix(DemoGo* this) {
...
     CutsceneContext* csCtx = &globalCtx->csCtx;
-    CsCmdActorAction* npcAction = csCtx->npcActions[func_8097C870(this)];
-    if (csCtx->state != 0) {
-        if (npcAction != NULL && npcAction->action == csCmdNPCAction) {
-            return 1;
-        }
+    s32 actionIdx = func_8097C870(this);
+
+    if ((csCtx->state != 0) && (csCtx->npcActions[actionIdx] != NULL) &&
+        (csCtx->npcActions[actionIdx]->action == csCmdNPCAction)) {
```
Comment (fig02) [inline src/overlays/actors/ovl_Demo_Go/z_demo_go.c:208]: common name for this temp is npcAction, works well imo. context that its from a cs command is obvious
I know its the name that was already there, just saying in general

Comment (fig02) [reply src/overlays/actors/ovl_Demo_Go/z_demo_go.c:208]: wait im confused, theres an npcAction temp and another one thats passed in?

Comment (fig02) [reply src/overlays/actors/ovl_Demo_Go/z_demo_go.c:208]: nvm im just seeing things. rename arg imo

Comment (ethteck) [reply src/overlays/actors/ovl_Demo_Go/z_demo_go.c:208]: kk yeah I thought the arg name was strange but I left it as is

**Pull request merged**